File: arduinobot_ws/src/arduinobot_test/src/simple_action_client.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <arduinobot_test/FibonacciAction.h>

int main (int argc, char **argv)
{
  ros::init(argc, argv, "fibonacci_client");

  // create the action client
  // true causes the client to spin its own thread
  actionlib::SimpleActionClient<arduinobot_test::FibonacciAction> ac("fibonacci");

  ROS_INFO("Waiting for action server to start.");
  // wait for the action server to start
  ac.waitForServer(); //will wait for infinite time

  ROS_INFO("Action server started, sending goal.");
  // send a goal to the action
  arduinobot_test::FibonacciGoal goal;
  goal.order = 20;
  ac.sendGoal(goal);

  //wait for the action to return
  ac.waitForResult(ros::Duration(30.0));

  arduinobot_test::FibonacciResultConstPtr result = ac.getResult();
  ROS_INFO("Action finished:");
  for (int i : result->sequence)
  {
    ROS_INFO("%d", i);
  }
  
  //exit
  return 0;
}

File: arduinobot_ws/src/arduinobot_test/scripts/simple_action_client.py
#! /usr/bin/env python3
import rospy
import actionlib
from arduinobot_test.msg import FibonacciAction, FibonacciGoal


def fibonacci_client():
    # Creates the SimpleActionClient, passing the type of the action
    # (FibonacciAction) to the constructor.
    client = actionlib.SimpleActionClient('fibonacci', FibonacciAction)

    # Waits until the action server has started up and started
    # listening for goals.
    client.wait_for_server()

    # Creates a goal to send to the action server.
    goal = FibonacciGoal(order=20)

    # Sends the goal to the action server.
    print("Seding goal order %s" % goal.order)
    client.send_goal(goal)

    # Waits for the server to finish performing the action.
    client.wait_for_result()

    # Prints out the result of executing the action
    return client.get_result()  


if __name__ == '__main__':
    # Inizialize a ROS node called fibonacci_client
    rospy.init_node('fibonacci_client')

    print("Simple Action Client Started")

    # Keep this node active until the action server returns a result
    result = fibonacci_client()
    print("Result: ", result.sequence)